src/postformatting/unwanted-properties-remover.js

```diff
@@ -1,41 +1,44 @@
-'use strict'
+import csstree from 'css-tree'
 
-const _removePropertiesFromRule = function (rule, propertiesToRemoveRegexes) {
-  if (
-    rule.type === 'Rule' ||
-    (rule.type === 'Atrule' && rule.name === 'font-face')
-  ) {
-    rule.block.children = rule.block.children.filter(declaration => {
-      if (
-        propertiesToRemoveRegexes.some(toRemovePattern => {
-          return toRemovePattern.test(declaration.property)
-        })
-      ) {
-        return false
-      }
-      return true
-    })
-  } else if (rule.type === 'Atrule' && rule.name === 'media') {
-    rule.block.children = rule.block.children
-      .map(function (rule) {
-        return _removePropertiesFromRule(rule, propertiesToRemoveRegexes)
-      })
-      .filter(Boolean)
-  }
-  if (rule.block && rule.block.children.length === 0) {
-    return null
-  }
-  return rule
-}
-
-const unwantedPropertiesRemover = function (astRules, propertiesToRemove) {
+const unwantedPropertiesRemover = function (ast, propertiesToRemove) {
   const propertiesToRemoveRegexes = propertiesToRemove.map(
     text => new RegExp(text)
   )
+  const shouldRemove = declaration =>
+    propertiesToRemoveRegexes.some(toRemovePattern =>
+      toRemovePattern.test(declaration.property)
+    )
+
+  csstree.walk(ast, {
+    visit: 'Declaration',
+    enter: (declaration, item, list) => {
+      if (shouldRemove(declaration)) {
+        list.remove(item)
+      }
+    }
+  })
+
+  // remove empty rules
+  csstree.walk(ast, {
+    visit: 'Rule',
+    leave: (rule, item, list) => {
+      if (rule.block.children.isEmpty()) {
+        list.remove(item)
+      }
+    }
+  })
+
+  // remove empty at-rules
+  csstree.walk(ast, {
```

Comment:
Why is this needed given the check you added [here](https://github.com/pocketjoso/penthouse/pull/222/files#diff-71b8f79184f85a105511f48046105fe2R165)? Perhaps because postformatting could have emptied the rule.. Hmm, ok with this though, not as expensive as walking every rule in the ast.